leaf-ui/js/objects.js

```diff
@@ -207,6 +207,7 @@ class AnalysisConfiguration {
 
 	constructor(id, analysisRequest) {
 		this.id = id;
+		this.analysisRequest = analysisRequest;
```

Comment:
thank you for updating this, its v clean! love it

Reply:
Yeah I figured since we ended up needing to grab the analysisRequest from the analysisConfig so often it might be easier to just save directly. Left in the other values for the Config class for now though. I think its mostly a design tradeoff between the efficiency of being able to save and get the whole request faster vs not storing more information from the request than necessary. AnalysisRequest currently contains the previous step value which we need for next states so once we get there we can probably discuss in more detail what parts of the Config class we need and what we don't.